classes/module/Module.php

```diff
@@ -2799,7 +2792,13 @@ public function addOverride($classname)
 
             $module_file = file($path_override);
             $module_file = array_diff($module_file, array("\n"));
-            eval(preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file)));
+            $new_override = preg_replace(array('#^\s*<\?(?:php)?#', '#class\s+'.$classname.'(\s+extends\s+([a-z0-9_]+)(\s+implements\s+([a-z0-9_]+))?)?#i'), array(' ', 'class '.$classname.'Override'.$uniq), implode('', $module_file));
+            try {
+                $parser->parse($new_override);
+            } catch (PhpParser\Error $e) {
+                throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
+            }
+            eval($new_override);
```

Comment:
I would add here this extra check in eval (for NULL result, if for some weird reason $parser->parse is tricked) or the code gets to rais ParseError in PHP 7.0+ (it is only a few more lines and a more secure extra check).

	try {
		if(NULL == eval($new_override)) {
			throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
		}
	} catch (ParseError $e) { // For PHP 7.0+ only will triggered
		throw new Exception(sprintf(Tools::displayError('The override file "%1$s" of the module %2$s is damaged or invalid.'), 'override'.DIRECTORY_SEPARATOR.$path, $this->displayName));
	}